diff --git a/CMakeLists.txt b/CMakeLists.txt
index e734bf723ef5..c82b50f5f4c0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 2.6...3.5)
 
 PROJECT(procd C)
 INCLUDE(GNUInstallDirs)
diff --git a/upgraded/CMakeLists.txt b/upgraded/CMakeLists.txt
index 09cf472e47ee..753cc63fb5f2 100644
--- a/upgraded/CMakeLists.txt
+++ b/upgraded/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 2.6...3.5)
 
 PROJECT(upgraded C)
 FIND_PATH(ubox_include_dir libubox/uloop.h)
